### PR TITLE
Improve error message when semgrep-core does not return results

### PR DESCRIPTION
**Pull request by emjin**
Add help suggestions to the error message. This also will make it easier to search for this specific error.

Test plan: I'm still trying to figure out how to force the error. I might just run it in docker.

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [ ] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: cli/src/semgrep/core_runner.py [hunk 1]
```diff
@@ -231,8 +231,19 @@ async def _core_stdout_processor(self, stream: asyncio.StreamReader) -> None:
                 # happens if the data that follows a sequence of zero
                 # or more ".\n" has fewer than two bytes, such as:
                 # "", "3", ".\n.\n3", ".\n.\n.\n.", etc.
+
+                # Hack: the exact wording of parts this message may be used in metrics queries
+                # that are looking for it. Make sure `semgrep-core exited with unexpected output`
```
Comment (nmote) [inline cli/src/semgrep/core_runner.py:236]: :grimacing:

**Review by nmote — APPROVED** [review]

**Comment by emjin** [comment]
Changed this to a draft. I'm going to check on how datadog monitoring works and might change this error according to what would make that easy

**Comment by emjin** [comment]
image

**Comment by IagoAbal** [comment]
> We recommend choosing a limit 70% of the available memory

How did we come up with this number?

File: cli/src/semgrep/core_runner.py [hunk 2]
```diff
@@ -231,8 +231,19 @@ async def _core_stdout_processor(self, stream: asyncio.StreamReader) -> None:
...
                 raise SemgrepError(
-                    "semgrep-core exited successfully but produced unexpected output"
+                    "semgrep-core exited with unexpected output.\n\n"
+                    "\tThis can happen because it overflowed the stack or used too much memory.\n"
+                    "\tTry changing the stack limit with `ulimit -s <limit>` or adding\n"
+                    "\t`--max-memory <memory>` to the semgrep command. For CI runs with interfile\n"
+                    "\tanalysis, the default max-memory is 5000MB. Depending on the memory\n"
+                    "\tavailable in your runner, you may need to lower it. We recommend choosing\n"
+                    "\ta limit 70% of the available memory to allow for some overhead. If you have\n"
+                    "\ttried these steps and still are seeing this error, please contact us."
```
Comment (ievans) [inline cli/src/semgrep/core_runner.py:246]: contact us at support@semgrep.dev? I think there is a constant in the codebase for it already

**Pull request merged**